navigation-fragments/src/main/java/dev/hotwire/navigation/navigator/NavigatorHost.kt

```diff
@@ -63,6 +71,22 @@ open class NavigatorHost : NavHostFragment(), FragmentOnAttachListener {
         }
     }
 
+    @VisibleForTesting(otherwise = PROTECTED)
+    fun ensureDeeplinkStartLocationValid() {
```

Comment:
I think some code documentation would be beneficial long-term. It'll be easy to forget why this is here. Something like:
```kotlin
/**
 * Google's Navigation library automatically navigates to deep links provided in the 
 * Activity's Intent. This exposes a vulnerability for malicious Intents to open an arbitrary 
 * webpage outside of the app's domain, allowing javascript injection on the page. Ensure 
 * that deep link intents always match the app's domain.
 */
```

Reply:
Thanks!